pkg/dataplane/deployment.go

```diff
@@ -82,6 +83,17 @@ func (d *Deployer) Deploy(services []string) (*ctrl.Result, error) {
 		nsConditions := d.Status.NodeSetConditions[d.NodeSet.Name]
 		log.Info("Deploying service", "service", service)
 		foundService, err := GetService(d.Ctx, d.Helper, service)
+
+		// For global services, check if it's already been deployed, and if so,
+		// skip deploying it.
+		if foundService.Spec.DeployOnAllNodeSets {
+			if !slices.Contains(d.GlobalServicesDeployed, service) {
+				d.GlobalServicesDeployed = append(d.GlobalServicesDeployed, service)
+			} else {
+				return &ctrl.Result{}, err
```

Comment:
I  think this should go after we handle the error  from L85 as foundService could be nil in case of error.  Also I guess we should skip the service rather than returning from here in case the `servicesOverride` has more services after the global service. Good to test that scenario too in functional tests.

Reply:
btw, I had a slightly different approach using the existing de-duplication logic https://github.com/openstack-k8s-operators/openstack-operator/compare/main...rabi:openstack-operator:override?expand=1, if thats of interest.

Reply:
maybe I'm not reading it right, but it looks like DedupeServices would result in NodeSetServiceMap values being empty for both NodeSets when servicesOverride contains a single global service.

then that empty value gets passed to Deploy in deployment_controller.

Reply:
Not really, it would pass empty for the second nodeset, unless you're not looking at code I linked in the diff. There is functional test for the scenario you mention in that and I proposed as a PR to get CI results.. https://github.com/openstack-k8s-operators/openstack-operator/pull/1029

Reply:
ok, i can see how that fixes the issue. Do you want to go with that instead?